solr/core/src/test/org/apache/solr/cli/ZkSubcommandsTest.java

```diff
@@ -398,66 +411,82 @@ public void testUpConfigLinkConfigClearZk() throws Exception {
-    zkFiles =
-        zkClient.getChildren(ZkConfigSetService.CONFIGS_ZKNODE + "/" + confsetname, null, true);
-    assertEquals(
-        "Comparing original conf files that were to be uploadedto what is in ZK",
-        files.length,
-        zkFiles.size());
-    assertEquals("Comparing downloaded files to what is in ZK", files.length, zkFiles.size());
+    Path confSetDir = configSetDir.resolve("conf");
+    try (Stream<Path> filesStream = Files.list(confSetDir)) {
+      List<Path> files = filesStream.toList();
+      zkFiles =
+          zkClient.getChildren(ZkConfigSetService.CONFIGS_ZKNODE + "/" + confsetname, null, true);
+      assertEquals(
+          "Comparing original conf files that were to be uploaded to what is in ZK",
+          files.size(),
+          zkFiles.size());
+      assertEquals("Comparing downloaded files to what is in ZK", files.size(), zkFiles.size());
+    }
 
-    File sourceConfDir = new File(ExternalPaths.TECHPRODUCTS_CONFIGSET);
+    Path sourceConfDir = ExternalPaths.TECHPRODUCTS_CONFIGSET;
     // filter out all directories starting with . (e.g. .svn)
-    Collection<File> sourceFiles =
-        FileUtils.listFiles(
-            sourceConfDir, TrueFileFilter.INSTANCE, new RegexFileFilter("[^\\.].*"));
-    for (File sourceFile : sourceFiles) {
-      int indexOfRelativePath =
-          sourceFile
-              .getAbsolutePath()
-              .lastIndexOf("sample_techproducts_configs" + File.separator + "conf");
-      String relativePathofFile =
-          sourceFile
-              .getAbsolutePath()
-              .substring(indexOfRelativePath + 33, sourceFile.getAbsolutePath().length());
-      File downloadedFile = new File(confDir, relativePathofFile);
-      if (ConfigSetService.UPLOAD_FILENAME_EXCLUDE_PATTERN.matcher(relativePathofFile).matches()) {
-        assertFalse(
-            sourceFile.getAbsolutePath()
-                + " exists in ZK, downloaded:"
-                + downloadedFile.getAbsolutePath(),
-            downloadedFile.exists());
-      } else {
-        assertTrue(
-            downloadedFile.getAbsolutePath()
-                + " does not exist source:"
-                + sourceFile.getAbsolutePath(),
-            downloadedFile.exists());
-        assertTrue(
-            relativePathofFile + " content changed",
-            FileUtils.contentEquals(sourceFile, downloadedFile));
-      }
+    try (Stream<Path> stream = Files.walk(sourceConfDir)) {
+      List<Path> files =
+          stream
+              .filter(Files::isRegularFile)
+              .filter(path -> path.getFileName().startsWith("."))
+              .toList();
+      files.forEach(
+          (sourceFile) -> {
+            int indexOfRelativePath =
+                sourceFile
+                    .toAbsolutePath()
+                    .toString()
+                    .lastIndexOf(
+                        "sample_techproducts_configs"
+                            + FileSystems.getDefault().getSeparator()
```

Comment:
Wherever we see `FileSystems.getDefault()` yet a Path var is in scope (here it's `sourceFile`), we should use `pathVar.getFileSystem()` instead.